src/untyped_array.rs

```diff
@@ -0,0 +1,254 @@
+//! Safe, untyped interface for NumPy's [N-dimensional arrays][ndarray]
+//!
+//! [ndarray]: https://numpy.org/doc/stable/reference/arrays.ndarray.html
+use std::{os::raw::c_int, slice};
+
+use pyo3::{
+    ffi, pyobject_native_type_extract, pyobject_native_type_named, AsPyPointer, IntoPy, PyAny,
+    PyNativeType, PyObject, PyTypeInfo, Python,
+};
+
+use crate::cold;
+use crate::dtype::PyArrayDescr;
+use crate::npyffi;
+
+/// A safe, untyped wrapper for NumPy's [`ndarray`][ndarray] class.
```

Comment:
Nit:
```suggestion
/// A safe, untyped, and unshaped wrapper for NumPy's [`ndarray`][ndarray] class.
```

Reply:
I am not happy about "unshaped". For me, "untyped" refers to both the element type `T` as well as the index type `D`. In addition, these arrays do have a shape in the sense that they provide the `.shape()` method. It is just that their dimensionality is not known at compile time but that is true for the `IxDyn` case as well.

So in summary, I would prefer to leave this to the following sentence explaining that neither `T` nor `D` are constrained at compile time.

Reply:
I see. Sounds fair.